src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

```diff
@@ -882,20 +947,24 @@ function discoverController(
     kbnUrl.change('/discover');
   };
 
-  $scope.updateDataSource = Promise.method(function updateDataSource() {
+  $scope.updateDataSource = Promise.method(() => {
```

Comment:
It would be nice to get rid of `Promise` here if it won't burthen you

Reply:
that would need some additional refactoring, but yes, should be removed .. like lot's of other stuff, ... in a future PR, if you don't mind :)